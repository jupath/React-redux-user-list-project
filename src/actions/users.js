
export const usersAreLoading = bool => ({
  type: 'USERS_ARE_LOADING',
  isLoading: bool,
});

export const setUsers = users => ({
  type: 'SET_USERS',
  users,
});

export const startSetUsers = url => async (dispatch) => {
  dispatch(usersAreLoading(true));

  try {
    const response = await fetch(url);
    const data = await response.json();
    const results = await data.results;
    const users = await results.map(user => ({
      username: user.login.username,
      firstName: user.name.first,
      lastName: user.name.last,
      email: user.email,
      address: `${user.location.street}, ${user.location.city}, ${user.location.state}, ${user.location.postcode}`,
      dob: user.dob,
      cell: user.cell,
      image: user.picture.large,
    }));
    await dispatch(usersAreLoading(false));
    await dispatch(setUsers(users));
  } catch (e) {
    console.error(e);
  }
};

export const updateUser = (username, updatedData) => ({
  type: 'UPDATE_USER',
  username,
  updatedData,
});
