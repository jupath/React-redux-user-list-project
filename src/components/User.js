import React from 'react';
import Parser from 'html-react-parser';
import { Link } from 'react-router-dom';

const User = (props) => {
  const {
    lastName, firstName, image, username, cell,
  } = props.userData;

  return (
    <div className="col-sm-4 mb-4">
      <div className="user p-3">
        <div>
          <img src={image} className="mr-2" />
        </div>
        <div className="user__text">
          <h5 className="text-capitalize font-weight-bold">{Parser(lastName)}, {Parser(firstName)}</h5>
          <p>Cell: {cell}</p>
          <div className="user__text__btnwrapper">
            <Link className="btn btn-success" to={`/user/${username}`}>Details</Link>
          </div>
        </div>
      </div>
    </div>
  );
};

export default User;
