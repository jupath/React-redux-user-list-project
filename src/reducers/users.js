export const usersAreLoading = (state = false, action) => {
  switch (action.type) {
    case 'USERS_ARE_LOADING':
      return action.isLoading;
    default:
      return state;
  }
};

export const users = (state = [], action) => {
  switch (action.type) {
    case 'SET_USERS':
      return action.users;
    case 'UPDATE_USER':
      return state.map((user) => {
        if (user.username === action.username) {
          return {
            ...user,
            ...action.updatedData,
          };
        }
        return user;
      });
    default:
      return state;
  }
};
