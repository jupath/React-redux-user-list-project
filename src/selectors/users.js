
export default (users, { text, orderBy }) => {
  return users.filter(user => {
    const regex = new RegExp( text, 'gi' );
    return user.firstName.match( regex ) || user.lastName.match( regex );
  }).sort((a, b) => {
    if( orderBy === 'desc' ) {
      return a.lastName < b.lastName ? 1 : -1;
    } else if ( orderBy === 'asc' ) {
      return a.lastName < b.lastName ? -1 : 1;
    }
  });
}