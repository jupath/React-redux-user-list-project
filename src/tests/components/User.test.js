import React from 'react';
import { shallow } from 'enzyme';
import User from '../../components/User';
import users from '../fixtures/users';

test('should render User correctly', () => {
  const wrapper = shallow(<User userData={users[0]} />);
  expect(wrapper).toMatchSnapshot();
});