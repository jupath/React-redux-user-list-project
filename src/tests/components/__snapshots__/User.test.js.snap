// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render User correctly 1`] = `
<div
  className="col-sm-4 mb-4"
>
  <div
    className="user p-3"
  >
    <div>
      <img
        className="mr-2"
        src="https://randomuser.me/api/portraits/men/83.jpg"
      />
    </div>
    <div
      className="user__text"
    >
      <h5
        className="text-capitalize font-weight-bold"
      >
        hoogmoed
        , 
        romain
      </h5>
      <p>
        Cell: 
        (065)-247-9303
      </p>
      <div
        className="user__text__btnwrapper"
      >
        <Link
          className="btn btn-success"
          replace={false}
          to="/user/lazyduck408"
        >
          Details
        </Link>
      </div>
    </div>
  </div>
</div>
`;
